src/main/java/com/anaplan/client/AnaplanService.java

```diff
@@ -0,0 +1,228 @@
+
+  private static final Logger LOG = LoggerFactory.getLogger(AnaplanService.class);
+
+  /**
+   * Sets metadata details for the connection with Anaplan server
+   *
+   * @param config the config instance carries metadata for the connection with Anaplan server
+   */
+  public static void setTaskMetadata(AnaplanConfig config) {
+    setAPIRoot(config.getServiceLocation(), config.getAuthServiceLocation());
+    setCredential(config.getUsername(), config.getPassword());
+  }
+
+  private static void setAPIRoot(String serviceLocation, String authServiceLocation) {
+    try {
+      Program.setServiceLocation(new URI(serviceLocation));
+      Program.setAuthServiceLocation(new URI(authServiceLocation));
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(e.getMessage());
+    }
+  }
+
+  private static void setCredential(String username, String password) {
+    Program.setUsername(username);
+    Program.setPassphrase(password);
+  }
+
+  /** Closes the connection with Anaplan server */
+  public static void closeDown() {
+    Program.closeDown();
+    LOG.info("Anaplan connection is closed.");
+  }
+
+  /**
+   * Runs the Anaplan functions defined in the model
+   *
+   * @param workspaceId the ID of the target workspace
+   * @param modelId the ID of the target model
+   * @param functionId the ID of the function to be run
+   * @param type the type {@link FunctionType} of the function to be run
+   * @throws InterruptedException when errors happens during the function run
+   */
+  public static void runAnaplanFunction(
+      String workspaceId, String modelId, String functionId, FunctionType type)
+      throws InterruptedException {
+    TaskFactory taskFactory = null;
+    switch (type) {
+      case PROCESS:
+        taskFactory = Program.getProcess(workspaceId, modelId, functionId);
+        if (taskFactory == null) {
+          throw new ProcessesNotFoundException(
+              modelId, new Exception("A process must be specified in model before calling"));
+        }
+        break;
+      case ACTION:
+        taskFactory = Program.getAction(workspaceId, modelId, functionId);
+        if (taskFactory == null) {
+          throw new ActionsNotFoundException(
+              modelId, new Exception("An action must be specified in model before calling"));
+        }
+        break;
+    }
+
+    Task task = taskFactory.createTask(new TaskParameters());
+    TaskResult lastResult = task.runTask();
+    LOG.info("Function run is done successfully:" + lastResult.isSuccessful());
+  }
+
+  /**
+   * Gets the {@link OutputStream} for uploading a data file
+   *
+   * @param workspaceId the ID of the target workspace
+   * @param modelId the ID of the target model
+   * @param fileId the ID of the placeholder in Anaplan model for the file to be loaded
+   * @param chunkSize the size of a chunk for uploading
+   * @return the {@link OutputStream} for uploading a data file
```

Comment:
javadoc should mention that it can throw ServerFilesNotFoundException

Reply:
Thanks, updated!